react/react1/week3/to-do-list-3/src/components/to-do-list.jsx

```diff
@@ -0,0 +1,300 @@
+          id: maxId + 1,
+          task: toDoTaskTxtBox,
+          deadLine: deadlineDate,
+          isCompleted: false,
+          isEditModeOn: false,
+        },
+      ]);
+    }
+  }
+
+  function DeleteTask(taskId) {
+    console.log("dlete " + taskId);
+    setToDoList(toDoList.filter((item) => item.id !== taskId));
+  }
+  function checkBoxStateChange(taskId) {
+    const taskArray = [...toDoList];
+    const currentTaskIndex = taskArray.indexOf(
+      taskArray.find((x) => x.id === taskId)
+    );
+    if (currentTaskIndex !== -1) {
+      taskArray[currentTaskIndex].isCompleted =
+        !taskArray[currentTaskIndex].isCompleted;
+    }
+    console.log(taskArray);
+    setToDoList(taskArray);
+  }
+
+  function editHandler(taskId) {
+    const taskArray = [...toDoList];
+    const currentTaskIndex = taskArray.indexOf(
+      taskArray.find((x) => x.id === taskId)
+    );
+    if (currentTaskIndex !== -1) {
+      taskArray[currentTaskIndex].isEditModeOn =
+        !taskArray[currentTaskIndex].isEditModeOn;
+    }
+    setToDoList(taskArray);
+  }
+
+  function handleTaskNameChange(taskName, taskId) {
+    console.log("handleTaskNameChange-", taskName, taskId);
+    const taskArray = [...toDoList];
+    const currentTaskIndex = taskArray.indexOf(
+      taskArray.find((x) => x.id === taskId)
+    );
+    if (currentTaskIndex !== -1) {
+      taskArray[currentTaskIndex].task = taskName;
+    }
+    setToDoList(taskArray);
+  }
+
+  return (
+    <>
+      <div className="container">
+        <label>To Do Description</label>
+        <input type="text" onChange={inputTextChangeHandler}></input>
+        <label>DeadLine</label>
+        <DatePicker selected={deadlineDate} onChange={dateChangeHandler} />
+        <label className="errorMsg">{taskNameErrorMsg}</label>
+        <span className="errorMsg">{dateErrorMsg}</span>
+        <button onClick={AddClickHandler}>Add todo</button>
+        <label>{toDoList.length} Items</label>
+      </div>
+
+      <ul className="task-list-ul">
+        {toDoList.map((taskItem) => (
+          <li key={taskItem.id}>
+            <BorderBox>
+              <TaskToDo
+                todoItem={taskItem}
+                handleTaskNameChange={(taskName, id) =>
+                  handleTaskNameChange(taskName, id)
+                }
+              />
+              <input
+                type={"checkbox"}
+                onChange={() => checkBoxStateChange(taskItem.id)}
+                checked={taskItem.isCompleted}
+              ></input>
+              <button onClick={() => DeleteTask(taskItem.id)}>Delete</button>
+              <button onClick={() => editHandler(taskItem.id)}>
+                {taskItem.isEditModeOn ? "Update" : "Edit"}
+              </button>
+            </BorderBox>
+          </li>
+        ))}
+      </ul>
+    </>
+  );
+}
+
+function TaskToDo(props) {
+  // console.log("To do Item-", props.todoItem);
+  const [toDoTaskNameTxtBox, updateToDoTaskNameTxtBox] = useState(
```

Comment:
I am not sure why you are storing the prop in state. You can just use the prop as is, there is no need to store it in state.